### PR TITLE
Use the same buffer information throughout file loading

### LINKED ISSUE #423: loading file with holes causes error
Reproduce step:
1. create a new file
2. write the following
   
   ``` idris
    test : String
    test = ?hole
   ```
3. run `idris-load-file`

Debugger info:

> Debugger entered--Lisp error: (wrong-type-argument stringp nil)
>   file-name-directory(nil)
>   idris-filename-to-load()
>   idris-update-loaded-region(nil)
>   #[257 "\302 \210\303 \210p\304   \235\203\305 \210\306\307!\210\310!\207" [idris-currently-loaded-buffer idris-warnings-printing idris-make-clean idris-update-options-cache warnings-tree idris-list-compiler-notes run-hooks idris-load-file-success-hook idris-update-loaded-region] 3 "\n\n(fn RESULT)"](nil)
>   #[257 "\300:\203/@A\301=\204!\202_\211:\203'\211@A\211\204\211\302:\203@\262\211:\203\211@A:\203
> @A:\203@A\303=\203 \211:\203  \211@A\211\204:\203@A:\203@A\304=\203\211:\203\211@A\211:\203\211@A\211\204\377:\203\377@A:\203\375@A\305=\203\373\211:\203\373\211@A\211:\203\371\211@A\211\204\367\204\367:\203\367@A\211\204\365\306!\306\307 !\230\203\363\310p   &\210\266\266\266\266\266\266\266\266\266\266\266\266\266\266A\262\202&\302\266\202\262\202\"!\266\202\202_!\266\202\2022\211!\207" [#[257 "\302 \210\303 \210p\304  \235\203\305 \210\306\307!\210\310!\207" [idris-currently-loaded-buffer idris-warnings-printing idris-make-clean idris-update-options-cache warnings-tree idris-list-compiler-notes run-hooks idris-load-file-success-hook idris-update-loaded-region] 3 "\n\n(fn RESULT)"] :highlight-source nil :filename :start :end file-name-nondirectory buffer-file-name idris-highlight-input-region] 48 "\n\n(fn RESULT)"](nil)
>   #[1028 "\211\211@A\300\301\"\2035\211\211G\302U\203\211@\202 \303\304\305GD\"J\2050Jq\210J!\266\202\202\204\300\306\"\203\200\211\211\203J\211A\262\242\202Q\303\304\305GD\"\211A\262\242\203f\303\304\305\307G\D\"\210    J\203wJq\210    J!\210\310\311\"\266\203\202\204\312\313\"\207" [eql :ok 1 signal wrong-number-of-arguments nil :error 2 message "Evaluation returned an error: %s." error "ELISP destructure-case failed: %S"] 15 "\n\n(fn G69925 G69926 G69927 G69921)"](--failure-cont-- --buffer-- --cont-- %28:ok nil%29)
>   apply(#[1028 "\211\211@A\300\301\"\2035\211\211G\302U\203\211@\202 \303\304\305GD\"J\2050Jq\210J!\266\202\202\204\300\306\"\203\200\211\211\203J\211A\262\242\202Q\303\304\305GD\"\211A\262\242\203f\303\304\305\307G\D\"\210  J\203wJq\210    J!\210\310\311\"\266\203\202\204\312\313\"\207" [eql :ok 1 signal wrong-number-of-arguments nil :error 2 message "Evaluation returned an error: %s." error "ELISP destructure-case failed: %S"] 15 "\n\n(fn G69925 G69926 G69927 G69921)"] --failure-cont-- --buffer-- --cont-- (:ok nil))
>   (lambda (&rest --cl-rest--) (apply (quote #[1028 "\211\211@A\300\301\"\2035\211\211G\302U\203\211@\202 \303\304\305GD\"J\2050Jq\210J!\266\202\202\204\300\306\"\203\200\211\211\203J\211A\262\242\202Q\303\304\305GD\"\211A\262\242\203f\303\304\305\307G\D\"\210  J\203wJq\210    J!\210\310\311\"\266\203\202\204\312\313\"\207" [eql :ok 1 signal wrong-number-of-arguments nil :error 2 message "Evaluation returned an error: %s." error "ELISP destructure-case failed: %S"] 15 "\n\n(fn G69925 G69926 G69927 G69921)"]) (quote --failure-cont--) (quote --buffer--) (quote --cont--) --cl-rest--))((:ok nil))
>   idris-dispatch-event((:return (:ok nil) 42) #<process Idris IDE support>)
>   idris-connection-available-input(#<process Idris IDE support>)
>   idris-output-filter(#<process Idris IDE support> "00001a(:set-prompt \"*hello\" 42)\n000016(:return (:ok ()) 42)\n")

### PR DESCRIPTION
**Pull request by jsoo1**
If popwin opens some buffer aside from the idris source buffer when the file has not finished loading, then the result of (buffer-file-name) will be wrong or nil when the callback for idris-load-file is reached.

Closes #423

### CODE REVIEW

File: idris-commands.el [hunk 1]
```diff
@@ -197,14 +209,13 @@ line."
...
-        (let* ((dir-and-fn (idris-filename-to-load))
-               (fn (cdr dir-and-fn))
-               (srcdir (car dir-and-fn)))
-          (setq idris-currently-loaded-buffer nil)
-          (idris-switch-working-directory srcdir)
-          (idris-delete-ibc t) ;; delete the ibc to avoid interfering with partial loads
+        (setq idris-currently-loaded-buffer nil)
+        (idris-switch-working-directory (idris-file-info-srcdir file-to-load))
+        (idris-delete-ibc t) ;; delete the ibc to avoid interfering with partial loads
+        (let ((fn (idris-file-info-filename file-to-load))
```
Comment (jsoo1) [inline idris-commands.el]: I could use some help with macros.

**Comment by jsoo1** [comment]
The tests breaking seem to be because `*idris-events*` is no longer created. Any idea why that might happen because of these changes?

**Comment by jsoo1** [comment]
Ah, I think it was due to #445.

**Comment by jsoo1** [comment]
Moved the travis fixes and test fixes to their own separate PRs.

**Comment by david-christiansen** [comment]
I have pretty low brain bandwidth right now. Can you say a few words on what the underlying issue is and how this fixes it?

**Comment by jsoo1** [comment]
Gonna cherry pick this onto a fresh branch.

**Pull request closed without merging**